official/vision/image_classification/resnet/common.py

```diff
@@ -331,6 +331,14 @@ def define_pruning_flags():
   flags.DEFINE_integer('pruning_frequency', 100,
                        'Frequency for pruning.')
 
+def define_clustering_flags():
+  """Define flags for clustering methods."""
+  flags.DEFINE_string('clustering_method', None,
+                      'None (no clustering) or selective_clustering.')
+  flags.DEFINE_integer('number_of_clusters', 256,
+                       'Number of clusters used in each layer.')
```

Comment:
The description of the `number_of_clusters` flags should clarify how configurable the model really is. I'm inclined to omit this flag since it's not meaningfully configurable (it doesn't affect the whole model) and someone probably would rather directly modify the code at that point. 

For `selective_clustering` flag, I'd describe it further, like `selective_clustering, which will cluster
the last three Conv2d layers of the model`.

Reply:
Done both. 
Tho I am slightly skeptical about not having the flag `number_of_clusters`, the code does look concise in your refactoring.